src/main/java/de/deepshore/kafka/Xsd2avroController.java

```diff
@@ -105,12 +111,96 @@ public HttpResponse convert(@Body XsdPack xsdpack) {
+    public HttpResponse getJava(@Body XsdPack xsdpack) {
+        File xsdFile;
+
+        if(!xsdpack.getXsd().startsWith(PREFIX) && !xsdpack.getXsd().startsWith("<xsd")){
+            return HttpResponse.unprocessableEntity();
+        }
+        if(!xsdpack.getXml().startsWith(PREFIX)){
+            return HttpResponse.unprocessableEntity();
+        }
+
+        try {
+            xsdFile = File.createTempFile("xsd", String.valueOf(LocalDateTime.now()));
+            Files.write(xsdpack.getXsd().getBytes(StandardCharsets.UTF_8), xsdFile);
+        } catch (IOException e) {
+            LOG.error("Error while creating temporary File for XSD", e);
+            return HttpResponse.serverError();
+        }
+
+        try(FromXml.Value transform = new FromXml.Value()) {
+
+
+            Map<String, String> transformConfig;
+
+            if("".equals(xsdpack.getXpathRecordKey())){
+                transformConfig = Map.of(
+                        SCHEMA_PATH_CONFIG, xsdFile.getAbsoluteFile().toURL().toString(),
+                        XJC_OPTIONS_STRICT_CHECK_CONFIG, XJC_OPTIONS_STRICT_CHECK_CONFIG_VALUE,
+                        PACKAGE_CONFIG, this.getClass().getPackageName()
+                );
+
+            } else {
+                transformConfig = Map.of(
+                        SCHEMA_PATH_CONFIG, xsdFile.getAbsoluteFile().toURL().toString(),
+                        XJC_OPTIONS_STRICT_CHECK_CONFIG, XJC_OPTIONS_STRICT_CHECK_CONFIG_VALUE,
+                        PACKAGE_CONFIG, this.getClass().getPackageName(),
+                        XPATH_FOR_RECORD_KEY, xsdpack.getXpathRecordKey()
+                );
+            }
+
+            transform.configure(transformConfig);
+
+            File tempFile = File.createTempFile("compressed-java", String.valueOf(LocalDateTime.now()));
+
+            try(final FileOutputStream fos = new FileOutputStream(tempFile);
+                ZipOutputStream zipOut = new ZipOutputStream(fos)) {
+                List<File> generatedSourceFiles = transform.getGeneratedSourceFiles();
+
+                generatedSourceFiles.forEach(srcFile -> {
+                    try(FileInputStream fis = new FileInputStream(srcFile)) {
+                        final ZipEntry zipEntry = new ZipEntry(srcFile.getName());
+                        zipOut.putNextEntry(zipEntry);
+                        byte[] bytes = new byte[1024];
+                        int length;
+                        while((length = fis.read(bytes)) >= 0) {
+                            zipOut.write(bytes, 0, length);
+                        }
+                    } catch (IOException e) {
+                        LOG.warn("Error zipping", e);
+                        return;
+                    }
+                });
+
+
+                return HttpResponse.ok(new StreamedFile(new FileInputStream(tempFile), new MediaType(MediaType.APPLICATION_OCTET_STREAM)))
+                        .header("Content-type", "application/octet-stream")
```

Comment:
Auch hier bitte die Micronaut-static benutzen

```suggestion
                        .header("Content-type", MediaType.APPLICATION_OCTET_STREAM)
```